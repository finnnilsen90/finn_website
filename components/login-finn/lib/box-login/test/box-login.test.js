const React = require('react');
const ReactTestUtils = require('react-dom/test-utils');
const renderer = require('react-test-renderer')
let { shallow, mount, render } = require('enzyme');

const Box = require('../box-login.js');
const Button_one = require('../../../../lib/button-one/button-one.js');

let newBox = new Box();

describe('component connection', () => {

    it('should be defined', () => {
        expect(Box).toBeDefined();
    })

    it('for button should be defined', () => {
        expect(Button_one).toBeDefined();
    })

})  
    
describe('<Box />', () => {

    const wrapper = render(<Box />, { attachTo: document.body });

    it('renders components', () => {
        expect(wrapper).toMatchSnapshot();
    });

})

describe('<Button_two/>', () => {
    const Button_test = shallow(<Button_one name='Login' href='./' type='submit' value='Login'/>, { attachTo: document.body })

    it('should render props', () => {
        expect(Button_test).toMatchSnapshot();
    })
})

describe('Login functionality', () => {

    const Login = render(<Box />, { attachTo: document.body })
    const input_user = Login.find('.input_user')
    const input_pass = Login.find('.input_pass')

    it('should submit username', () => {
        expect(input_user.length).toBe(1);
    })

    it('should submit pass', () => {
        expect(input_pass.length).toBe(1);
    })
})

describe('Login error reporting', () => {

    it('should throw error if blank', () => {
        expect().toBeDefined()
    })

}