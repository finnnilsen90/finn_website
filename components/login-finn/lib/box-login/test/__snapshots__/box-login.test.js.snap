// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Box /> renders components 1`] = `
<div
  class="box-login_Container font"
>
  <div
    class="box"
  >
    <form
      action="/login"
      class="box_form"
      method="post"
    >
      <div
        class="input_box"
      >
        <label
          class="username label font"
        >
          Username
        </label>
        <input
          class="input_user"
          name="username"
          type="text"
        />
      </div>
      <div
        class="input_box"
      >
        <label
          class="password label font"
        >
          Password
        </label>
        <input
          class="input_pass"
          name="password"
          type="password"
        />
      </div>
      <div
        class="button"
      >
        <div
          class="button-one_Container"
        >
          <button
            class="button_style_one"
            href="./"
            type="submit"
            value="Login"
          >
            Login
          </button>
        </div>
      </div>
      <div
        class="request"
      >
        <a
          class="font"
          href="/"
        >
          Request Login
        </a>
      </div>
    </form>
  </div>
</div>
`;

exports[`<Button_two/> should render props 1`] = `
<div
  className="button-one_Container"
>
  <button
    className="button_style_one"
    href="./"
    type="submit"
    value="Login"
  >
    Login
  </button>
</div>
`;
